src/timeline/TimelineModel.cpp

```diff
@@ -741,6 +774,44 @@ TimelineModel::internalAddEvents(
 
                         if (encInfo)
                                 emit newEncryptedImage(encInfo.value());
+
+                        if (emitCallEvents) {
+                                if (auto callInvite = std::get_if<
+                                      mtx::events::RoomEvent<mtx::events::msg::CallInvite>>(&e_)) {
+                                        callInvite->room_id = room_id_.toStdString();
+                                        emit newCallEvent(e_);
+                                } else if (std::holds_alternative<mtx::events::RoomEvent<
+                                             mtx::events::msg::CallCandidates>>(e_) ||
+                                           std::holds_alternative<
+                                             mtx::events::RoomEvent<mtx::events::msg::CallAnswer>>( e_) ||
+                                           std::holds_alternative<
+                                             mtx::events::RoomEvent<mtx::events::msg::CallHangUp>>( e_)) {
+                                        emit newCallEvent(e_);
+                                }
+                        }
+                }
+
+                if (std::holds_alternative<
+                      mtx::events::RoomEvent<mtx::events::msg::CallCandidates>>(e)) {
```

Comment:
I think some events may be encrypted at this point, so they still get added to the timeline in this case. You probably want to access the decrypted event here too for reference.

Reply:
Agreed, fixed.